dockers/docker-dhcp-server/cli/config/plugins/dhcp_server.py

```diff
@@ -1,13 +1,371 @@
+    dbconn.set("CONFIG_DB", key, "state", "enabled")
+
+
+@ipv6.command()
+@click.argument("dhcp_interface", required=True)
+@clicommon.pass_db
+def enable(db, dhcp_interface):
+    pass
+
+
+@ipv4.command()
+@click.argument("dhcp_interface", required=True)
+@clicommon.pass_db
+def disable(db, dhcp_interface):
+    dbconn = db.db
+    key = "DHCP_SERVER_IPV4|" + dhcp_interface
+    dbconn.set("CONFIG_DB", key, "state", "disabled")
+
+
+@ipv6.command()
+@click.argument("dhcp_interface", required=True)
+@clicommon.pass_db
+def disable(db, dhcp_interface):
+    pass
+
+
+@ipv4.command()
+@clicommon.pass_db
+def update(db):
+    pass
+
+
+@ipv6.command()
+@clicommon.pass_db
+def update(db):
+    pass
+
+
+@ipv4.group(cls=clicommon.AliasedGroup)
+def range():
+    pass
+
+
+@range.command()
+@click.argument("range_name", required=True)
+@click.argument("ip_start", required=True)
+@click.argument("ip_end", required=False)
+@clicommon.pass_db
+def add(db, range_name, ip_start, ip_end):
+    if not ip_end:
+        ip_end = ip_start
+    ipaddress.ip_address(ip_start)
+    ipaddress.ip_address(ip_end)
+    dbconn = db.db
+    key = "DHCP_SERVER_IPV4_RANGE|" + range_name
+    db.hmset("CONFIG_DB", key, {"ranges": ip_start + "," + ip_end})
+
+
+@range.command()
+@click.argument("range_name", required=True)
+@click.argument("ip_start", required=True)
+@click.argument("ip_end", required=False)
+@clicommon.pass_db
+def update(db, range_name, ip_start, ip_end):
+    if not ip_end:
+        ip_end = ip_start
+    ipaddress.ip_address(ip_start)
+    ipaddress.ip_address(ip_end)
+    dbconn = db.db
+    key = "DHCP_SERVER_IPV4_RANGE|" + range_name
+    db.hmset("CONFIG_DB", key, {"ranges": ip_start + "," + ip_end})
+
+
+@range.command()
+@click.argument("range_name", required=True)
+@clicommon.pass_db
+def del(db, range_name)
+    dbconn = db.db
+    key = "DHCP_SERVER_IPV4_RANGE|" + range_name
+    db.delete("CONFIG_DB", key)
+
+
+@ipv6.group(cls=clicommon.AliasedGroup)
+def range():
+    pass
+
+
+@range.command()
+@click.argument("range_name", required=True)
+@click.argument("ip_start", required=True)
+@click.argument("ip_end", required=False)
+@clicommon.pass_db
+def add(db, range_name, ip_start, ip_end):
+    pass
+
+
+@range.command()
+@click.argument("range_name", required=True)
+@click.argument("ip_start", required=True)
+@click.argument("ip_end", required=False)
+@clicommon.pass_db
+def update(db, range_name, ip_start, ip_end):
+    pass
+
+
+@range.command()
+@click.argument("range_name", required=True)
+@clicommon.pass_db
+def del(db, range_name)
+    pass
+
+
+@ipv4.group(cls=clicommon.AliasedGroup)
+def ip():
+    pass
+
+
+@ip.command()
+@click.argument("vlan_interface", required=True)
+@click.argument("interface", required=True)
+@click.options("--range", required=False, nargs=-1)
+@click.argument("ip_list", required=False, nargs=-1)
+@clicommon.pass_db
+def bind(db, vlan_interface, interface, range_, ip_list):
+    dbconn = db.db
```

Comment:
Check whether key exist.
If exist: incremental binding, merge old and new. Need to clarify that **cannot bind both ips and ranges in the same time**
If not exist: add new entry